cpp/open3d/geometry/TriangleMesh.cpp

```diff
@@ -77,30 +75,30 @@ TriangleMesh &TriangleMesh::operator+=(const TriangleMesh &mesh) {
     if (HasAdjacencyList()) {
         ComputeAdjacencyList();
     }
-    if (add_textures) {
+    if (mesh.HasTriangleUvs() && (HasTriangleUvs() || is_empty)) {
         size_t old_tri_uv_num = triangle_uvs_.size();
         triangle_uvs_.resize(old_tri_uv_num + mesh.triangle_uvs_.size());
         for (size_t i = 0; i < mesh.triangle_uvs_.size(); i++) {
             triangle_uvs_[old_tri_uv_num + i] = mesh.triangle_uvs_[i];
         }
-
+    }
+    if (mesh.HasTextures() && (HasTextures() || is_empty)) {
         size_t old_tex_num = textures_.size();
         textures_.resize(old_tex_num + mesh.textures_.size());
         for (size_t i = 0; i < mesh.textures_.size(); i++) {
             textures_[old_tex_num + i] = mesh.textures_[i];
         }
-
+    }
+    if (mesh.HasTriangleMaterialIds() &&
+        (HasTriangleMaterialIds() || is_empty)) {
+        size_t old_tex_num = textures_.size();
         size_t old_mat_id_num = triangle_material_ids_.size();
         triangle_material_ids_.resize(old_mat_id_num +
                                       mesh.triangle_material_ids_.size());
         for (size_t i = 0; i < mesh.triangle_material_ids_.size(); i++) {
             triangle_material_ids_[old_mat_id_num + i] =
                     mesh.triangle_material_ids_[i] + (int)old_tex_num;
         }
-    } else {
-        triangle_uvs_.clear();
-        textures_.clear();
-        triangle_material_ids_.clear();
     }
```

Comment:
We need to keep this portion - `clear()` properties if only one mesh has them (unless the other mesh is empty), else we will get meshes with only half the triangles with material ids, etc. (e.g. if `this` has material_ids, but `mesh` doesn't, output should not have material ids.) Half populated properties will cause seg faults.

Also, textures and triangle_material_ids go together, so we should have a single `if()` for them. We want `n_textures == max(triangle_material_ids_)+1` and either both or none should be present always. 

@cdbharath this PR may break other mesh functions if there is a bug -- please do the following tests and make sure there are no errors. Check for the case where:

- a has texture_uvs but no textures / triangle_material_ids
- b has all of texture_uvs, textures and triangle_material_ids 
- c has none of the 3 (but has triangles)

With these 3 meshes, perform this check:

- `a+=b` and `b+=a` should produce the same results in a and b respectively (test mesh equality). Also for the pair `a+=c`, `c+=a`, and separately for the pair `b+=c` and `c+=b`. [Add this to C++ unit tests if not present already] Visualize the resulting mesh correctly with draw_geometries() and draw() [Just test visualization on your computer - no unit tests required]